composer/utils/object_store/object_store_hparams.py

```diff
@@ -210,7 +214,40 @@ def get_kwargs(self) -> Dict[str, Any]:
         return dataclasses.asdict(self)
 
 
+@dataclasses.dataclass
+class SFTPObjectStoreHparams(ObjectStoreHparams):
+    """:class:`~.SFTPObjectStore` hyperparameters.
+
+    The :class:`.SFTPObjectStore` uses :mod:`paramiko` to upload and download files from SSH servers
+
+    .. note::
+
+        To follow best security practices, credentials shouldn't be specified as part of the hyperparameters.
+        Instead, please ensure that credentials are in the environment, which will be read automatically.
+
+    Args:
+        host (str): See :class:`.SFTPObjectStore`.
+        port (int, optional): See :class:`.SFTPObjectStore`.
+        username (str, optional): See :class:`.SFTPObjectStore`.
+        key_file_path (str, optional): See :class:`.SFTPObjectStore`.
+        cwd (str, optional): See :class:`.SFTPObjectStore`.
+    """
+
+    host: str = hp.auto(SFTPObjectStore, 'host')
+    port: int = hp.optional(('port'), default=22)
```

Comment:
For all these arguments, is it possible to use `hp.auto`? https://docs.mosaicml.com/projects/yahp/en/latest/api_ref/field.html#yahp.field.auto